ndt7.go

```diff
@@ -170,10 +185,32 @@ func (c *Client) start(ctx context.Context, f testFn, p string) (<-chan spec.Mea
 		return nil, err
 	}
 	ch := make(chan spec.Measurement)
-	go f(ctx, conn, ch)
+	go c.collectData(ctx, f, conn, ch)
 	return ch, nil
 }
 
+func (c *Client) collectData(ctx context.Context, f testFn, conn websocketx.Conn, outch chan<- spec.Measurement) {
```

Comment:
When we were discussing in private this file's implementation, you mentioned that you'd rather see this piece of code written using interfaces rather than in this function-oriented JavaScript-like fashion. Maybe open an issue?

Reply:
I would like to revisit my opinion about this function-oriented approach after writing the same summary for the ndt5-client (which, AFAIK, is more OO-like?). At the moment I don't feel strongly enough about this to suggest writing it differently.

Reply:
Sure. I should actually learn to mind my business more. It's natural that the coding style conventions change when the maintainer of the code changes, and this has already happened. I hope you find ndt5 less functional.